app/src/main/java/com/ncnf/event/Social.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+
+    private final String path;
+
+    private UUID uuid;
+    private String ownerId;
+    private String name;
+    private Date date;
+    private Type type;
+    private List<String> attendees;
+    private int numOfAttendees;
+    private String description;
+    private GeoPoint location;
+    private String address;
+
+    public Social(String ownerId, String name, Date date, GeoPoint location, String address, Type type, String description) {
+        this(ownerId, UUID.randomUUID(), name, date, location, address, type, new ArrayList<>(), description);
+    }
+
+    public Social(String ownerId, UUID id, String name, Date date, GeoPoint location, String address, Type type, List<String> attendees, String description) {
+        this.uuid = id;
+        this.path = EVENTS_COLLECTION_KEY + uuid;
+        this.ownerId = ownerId;
+        this.name = name;
+        this.date = date;
+        this.location = location;
+        this.address = address;
+        this.type = type;
+        this.attendees = attendees;
+        this.numOfAttendees = attendees.size();
+        this.description = description;
+    }
+
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Date getDate() {
+        return date;
+    }
+
+    public GeoPoint getLocation() {
+        return location;
+    }
+
+    public java.lang.String getAddress() {
+        return address;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public List<String> getAttendees() {
+        return Collections.unmodifiableList(attendees);
+    }
+
+    public int getNumOfAttendees() {
+        return numOfAttendees;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getOwnerId() {
+        return ownerId;
+    }
+
+    public void setUuid(UUID uuid) {
+        this.uuid = uuid;
+    }
+
+    public void setOwnerId(String ownerId) {
+        this.ownerId = ownerId;
+    }
+
+    public void setType(Type type) {
+        this.type = type;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setDate(Date date) {
+        this.date = date;
+    }
+
+    public void setLocation(GeoPoint location) {
+        this.location = location;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public void setAttendees(List<String> attendees) {
+        this.attendees = new ArrayList<>(attendees);
+        numOfAttendees = attendees.size();
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    abstract public CompletableFuture<Boolean> store(DatabaseService db);
+
+    public static CompletableFuture<Boolean> addNews(DatabaseService db, String uuid, String value) {
+       return db.updateArrayField(EVENTS_COLLECTION_KEY + uuid, NEWS_KEY, value);
+    }
```

Comment:
Should not be here.

Reply:
Yes I've moved it in the DBService class

Reply:
> Good refactoring ! We should probably not call what we used to call Events "socials" in the UI though since it's not a noun and the user might get confused, but that's something that we can fix later :)

I renamed it "SocialObject" : it makes more sense.

> I approve the new structure for events and groups that inherits from Social.
> However, in the views (create, view, adapter, ...), you use the Social class, which means the same view can be used for Event and Group, but what happens when they have specific fields ?
For some views, we act the same way on both sub-classes so it's not important but for others we might create two views, one for each sub-class so that it acts exactly the way we want to given a certain data type.

Reply:
I misunderstood what you meant : for now we will keep it here and once we are done with event creation and all we will see how to update event news etc.